Make HTTPMetricHandler class public

bernd opened the pull request:
This makes the HTTPMetricHandler reusable to allow custom HTTPServer implementations.

Fixes #529

@fstab @tomwilkie This is a replacement PR for the closed https://github.com/prometheus/client_java/pull/530. The #529 issue has been closed already but having this class exposed would help me with a custom `HTTPServer` implementation. :smile: Thank you!

fstab commented:
Looks good. Could you rebase it to the `next-release` branch? Then I'll merge it and it will be in the next release.

bernd commented:
> Looks good. Could you rebase it to the `next-release` branch? Then I'll merge it and it will be in the next release.

@fstab I rebased the PR to the `next-release` branch and changed the target branch as well! Thank you! :slightly_smiling_face:

fstab reviewed (approved):
Looks good to me. Thanks.
